### PR TITLE
:sparkles: add assign grammar

### PR DESCRIPTION
**Pull request by zwpaper**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

https://github.com/kcl-lang/tree-sitter-kcl/issues/2

#### 2. What is the scope of this PR (e.g. component or file name):
tree sitter kcl assign

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -34,7 +35,55 @@ module.exports = grammar({
...
+      "**=",
+      "/=",
+      "//=",
+      "%=",
+      "&=",
+      "|=",
+      "^=",
+      "<<=",
+      ">>=",
+      "or",
```
Comment (Peefy) [inline grammar.js:72]: `or` and `and` may not be right aug assign operators. And the operator precedence document is here: https://kcl-lang.io/docs/reference/lang/spec/kcl-spec/#operator-precedence

There may mistakes for specific grammar spec. U can ref https://kcl-lang.io/docs/reference/lang/spec/kcl-spec#grammar

Comment (zwpaper) [reply grammar.js:72]: Oops, let me fix it but then it seems that there may be some mistakes [here](https://kcl-lang.io/docs/reference/lang/spec/statements/#assignment-statements)? as I copied the list from there to save my time...

image

**Review by Peefy — APPROVED** [review]
LGTM

File: grammar.js (hunk 1 above)
Comment (Peefy) [reply grammar.js]: > Oops, let me fix it but then it seems that there may be some mistakes [here](https://kcl-lang.io/docs/reference/lang/spec/statements/#assignment-statements)? as I copied the list from there to save my time...
> 
> image

Yes. I'll fix and double check them. U can ref the total grammar: https://kcl-lang.io/docs/reference/lang/spec/kcl-spec#grammar

**Pull request merged**